### PR TITLE
Add node troubleshooting documentation

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Documents some tools that are available in our code base that can help
understanding why nodes are not able to join a cluster.

Also adds some fixes to the getting started documentation.

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] This change includes docs.

### CODE REVIEW

File: docs/content/how-to/debug-nodes.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    oc get machinedeployment -n $CONTROL_PLANE_NAMESPACE
+    oc get machineset -n $CONTROL_PLANE_NAMESPACE
+    ```
+    In the case that no machinedeployment has been created look at the logs of the hypershift 
+    opertor:
+    ```
+    oc logs deployment/operator -n hypershift
+    ```
+
+    If the mahines exist but have not been provisioned, check the log of the cluster API provider:
```
Comment (enxebre) [inline docs/content/how-to/debug-nodes.md:26]: typo mahines -> machines

File: docs/content/how-to/debug-nodes.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   oc get awsmachines -n $CONTROL_PLANE_NAMESPACE
+   ```
+
+    If machines don't exist, check that a machinedeployment and machineset have been created:
+    ```
+    oc get machinedeployment -n $CONTROL_PLANE_NAMESPACE
+    oc get machineset -n $CONTROL_PLANE_NAMESPACE
+    ```
+    In the case that no machinedeployment has been created look at the logs of the hypershift 
+    opertor:
```
Comment (enxebre) [inline docs/content/how-to/debug-nodes.md:21]: typo opertor -> operator

**Comment by csrwng** [comment]
Thanks @enxebre, addressed your comments

**Comment by sjenning** [comment]
Some nits, but no point in letting this linger
/lgtm

**Pull request merged**